ffi/flake.nix

```diff
@@ -0,0 +1,119 @@
+{
+  # To use:
+  #  - Install nix: https://github.com/DeterminateSystems/nix-installer?tab=readme-ov-file#install-nix
+  #  - Run from ffi/ dir: `nix build .#firewood-ffi`
+  #  - Run from anywhere: `nix develop 'github:ava-labs/firewood?dir=ffi&ref=[SHA]'`
+
+  description = "Firewood FFI library and development environment";
+
+  inputs = {
+    nixpkgs.url = "https://flakehub.com/f/NixOS/nixpkgs/0.2505.*.tar.gz";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+    crane.url = "github:ipetkov/crane";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, rust-overlay, crane, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+    let
+      overlays = [ (import rust-overlay) ];
+      pkgs = import nixpkgs { inherit system overlays; };
+      inherit (pkgs) lib;
+
+      rustToolchain = pkgs.rust-bin.stable.latest.default.override {
+        extensions = [ "rust-src" "rustfmt" "clippy" ];
+      };
+
+      craneLib = (crane.mkLib pkgs).overrideToolchain rustToolchain;
+
+      # Extract crate info from Cargo.toml files
+      ffiCargoToml = builtins.fromTOML (builtins.readFile ./Cargo.toml);
+      workspaceCargoToml = builtins.fromTOML (builtins.readFile ../Cargo.toml);
+
+      src = lib.cleanSourceWith {
+        src = craneLib.path ./..;
+        filter = path: type:
+          (lib.hasSuffix "\.md" path) ||
+          (craneLib.filterCargoSources path type);
+      };
+
+      commonArgs = {
+        inherit src;
+        strictDeps = true;
+        dontStrip = true;
+
+        # Build only the firewood-ffi crate
+        pname = ffiCargoToml.package.name;
+        version = workspaceCargoToml.workspace.package.version;
+
+        CARGO_PROFILE = "maxperf";
```

Comment:
this doesn't appear to be working. the size reduction @rkuris noticed is from fewer inlined methods which tells me there's less LTO. 

I haven't determined the correct way to set this with `rust-overlay`, but cargo will use the [`PROFILE`](https://doc.rust-lang.org/cargo/reference/environment-variables.html#:~:text=PROFILE%20%E2%80%94%20release,settings%20being%20used.) environment variable if `--profile` is not provided on the command line.

Reply:
Ah, I didn't realize the change to not strip came after Ron's comment.